cit3.0-web/src/components/Page/UserStory/UserStory.js

```diff
@@ -120,47 +144,73 @@ export default function UserStory() {
     }
 
     const userOption = userStoryPaths.find((x) => x.code === param.code);
+    // skip yes and no on the short version
+    if (!isLongVersion && userOption && userOption.group !== "zone") {
+      const isYes = userOption.user_story_paths.find((x) =>
+        x.code.includes("-YES")
+      );
+      if (isYes) {
+        userOption.user_story_paths = [isYes];
+      }
+    }
 
-    if (param.group === "area") {
+    if (param.group === "zone") {
       setAreaType(userOption.label);
-      setAreaFilterId(userOption.code.toLowerCase());
+      zoneType.current = userOption.code.toLowerCase();
     }
-
+    // zone list
     if (userOption.code.includes("REGIONALDISTRICTS")) {
       userOption.user_story_paths = regionals;
+      zoneType.current = "Regional Districts";
     }
 
     if (userOption.code.includes("COMMUNITYAREA")) {
       userOption.user_story_paths = communities;
     }
 
-    let replaceText = userOption.preText;
+    let replaceText = userOption.longText;
     replaceText = replaceText.replace(
-      "{AREA-TYPE-1}",
+      "{ZONE-TYPE-1}",
       userOption.label.slice(0, -1)
     );
-    replaceText = replaceText.replace("{AREA-TYPE-2}", areaType);
-    replaceText = replaceText.replace("{AREA-SEARCH-FILTER}", param.label);
-    userOption.postText = replaceText;
+    replaceText = replaceText.replace("{ZONE-TYPE-2}", areaType);
+    replaceText = replaceText.replace("{ZONE-SEARCH-FILTER}", param.label);
+    userOption.longTextLabel = replaceText;
+
+    setUserOptions([...newUserOptions, userOption]);
 
-    setAllOptions([...newUserOptions, userOption]);
+    if (param.code.includes("-YES") && isLongVersion) {
+      setIsOkButton(false);
+      showResult(param.url);
+      return;
+    }
 
     const isLastOption = userOption.user_story_paths.length < 2;
 
-    if (userOption.code.includes("-YES") || isLastOption) {
-      setIsYesButton(true);
-      setIsNoButton(true);
-      setRedirectURL(param.url);
+    if (userOption.code.includes("-GO") || isLastOption) {
+      setIsGoButton(true);
+      setIsOkButton(false);
+      redirectUrl.current = param.url;
     } else {
-      setIsYesButton(false);
-      setIsNoButton(false);
+      setIsGoButton(false);
+      setIsOkButton(false);
     }
+
     if (userOption.code.includes("-NO")) {
```

Comment:
should this be removed?

Reply:
so, if NO and the next path is also NO, it means end of path. There is a GENERIC-NO path which leads to END.
some NO answers will go to another story path.
and yeah, there is a long and version version where YES and NO have a different behavior.